client/src/App.js

```diff
@@ -1,24 +1,62 @@
-import React, { Fragment, Component } from 'react';
+import React, { Component } from 'react';
+import { decode } from 'jsonwebtoken';
 
 import Navbar from './components/Navbar';
 import Home from './components/Home';
+import UserContext from './utils/user-context';
 
 class App extends Component {
   state = {
-    loggedInUser: null,
+    user: null,
   };
 
+  /* eslint-disable no-console, react/destructuring-assignment */
   componentDidMount() {
-    // get logged in user
+    const token = localStorage.getItem('token');
+    if (!this.state.user && token) {
+      this.loginUsingJWT(token);
+    }
   }
 
+  loginUsingJWT = (token) => {
+    const decoded = decode(token);
+    const tokenIsValid = decoded !== null;
+    if (tokenIsValid) {
```

Comment:
Try to use early return pattern
```suggestion
    if (!tokenIsValid) {
       this.removeTokenFromLocalStorage();
       return;
    }
     const expired = decoded.exp <= new Date() / 1000;
     if (expired) {
         this.removeTokenFromLocalStorage();
         return; 
     }
     this.login({ id: decoded.id, username: decoded.username });
```